kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaDeletionService.java

```diff
@@ -0,0 +1,147 @@
+package com.slack.kaldb.clusterManager;
+
+import static com.google.common.util.concurrent.Futures.addCallback;
+import static com.slack.kaldb.config.KaldbConfig.DEFAULT_ZK_TIMEOUT_SECS;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.slack.kaldb.metadata.cache.CacheSlotMetadataStore;
+import com.slack.kaldb.metadata.replica.ReplicaMetadataStore;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import com.slack.kaldb.util.FutureUtils;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Deletes expired and unassigned replicas. If a replica is still assigned, marked for eviction, or
+ * in the process of evicting this will not delete the replica. Any replicas that are not deleted in
+ * a given run will be attempted in the following execution.
+ */
+public class ReplicaDeletionService extends AbstractScheduledService {
```

Comment:
Is there a reason to keep replica deletion and eviction a separate service? Isn't it more efficient to keep them in the same service?

Reply:
As discussed in more depth [on the eviction service pr](https://github.com/slackhq/kaldb/pull/179#discussion_r779724751) we're keeping these separate due to duration it takes to evict, and for better fault tolerance.